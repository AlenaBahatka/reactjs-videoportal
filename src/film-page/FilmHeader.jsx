import React from 'react';

function FilmHeader(props) {
    let {title, year, coverPicture, rating, description, fullDescription, duration} = props.filmInfo;
    return (
        <div className="row">
            <div className="col-sm-6">
                <img src={coverPicture} alt={title}></img>
            </div>
            <div className="col-sm-6">
                <h2>{title} <span>{rating}</span></h2>
                <span> {description} </span>
                <span> {year} </span>
                <span> {duration} </span>
                <div>
                    {fullDescription}
                </div>
            </div>

        </div>
    );
}

export default FilmHeader;