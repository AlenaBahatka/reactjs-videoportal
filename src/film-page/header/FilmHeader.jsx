import React from 'react';

function FilmHeader(props) {
	let { title, year, coverPicture, rating, description, fullDescription, duration } = props.filmInfo;
	return (
		<div className="row">
			<div className="col-sm-6">
				<img src={coverPicture} alt={title} />
			</div>
			<div className="col-sm-6" id="filmAdditionalInfo">
				<h2>
					{title} <span>{rating}</span>
				</h2>
				<span id="filmDescription"> {description} </span>
				<span id="filmYear"> {year} </span>
				<span id="filmDuration"> {duration} </span>
				<div id="filmDescription">{fullDescription}</div>
			</div>
		</div>
	);
}

export default FilmHeader;
