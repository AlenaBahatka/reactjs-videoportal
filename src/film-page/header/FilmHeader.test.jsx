import React from 'react';
import { shallow } from 'enzyme';
import FilmHeader from './FilmHeader';

describe('FilmHeader', () => {
	it('should correctly render component', () => {
		let filmInfo = {
            title: 'Roma',
            director: 'Alfonso Cuar√≥n',
            genre: 'comedy',
            year: 2017,
            coverPicture: 'pic.jpg',
            rating: 8,
            description: 'dddd',
            fullDescription: 'ddddddd',
            duration: 124,
            id: 'id1'
        };
        
        let filmHeaderComponent = shallow(<FilmHeader filmInfo={filmInfo} />);
        
		expect(filmHeaderComponent).toMatchSnapshot();
	});
});
