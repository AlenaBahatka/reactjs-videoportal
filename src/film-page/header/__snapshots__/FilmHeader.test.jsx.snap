// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmHeader should correctly render component 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-6"
  >
    <img
      alt="Roma"
      src="pic.jpg"
    />
  </div>
  <div
    className="col-sm-6"
    id="filmAdditionalInfo"
  >
    <h2>
      Roma
       
      <span>
        8
      </span>
    </h2>
    <span
      id="filmDescription"
    >
       
      dddd
       
    </span>
    <span
      id="filmYear"
    >
       
      2017
       
    </span>
    <span
      id="filmDuration"
    >
       
      124
       
    </span>
    <div
      id="filmDescription"
    >
      ddddddd
    </div>
  </div>
</div>
`;
